### PR TITLE
Added new routes for Deleted ManagedHsm for Soft Delete Feature

### PR DESCRIPTION
**Pull request by kim-soohwan**
…Hsm definition to include PurgeDateTimeUTC

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json [hunk 1]
```diff
@@ -349,6 +349,152 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Retrieved information about the specified deleted managed HSM.",
+            "schema": {
+              "$ref": "#/definitions/ManagedHsm"
+            }
+          },
+          "204": {
```
Comment (heaths) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json:427]: Per guidelines, this should be 404. But is this already consistent with Key Vault? If so, I'd leave it for consistency's sake.

Comment (heaths) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json:427]: No, this is not consistent with Key Vault: https://github.com/Azure/azure-rest-api-specs/blob/7256c18a35090d545a5a3595920c2275d3e397b3/specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keyvault.json#L596

This should return 404/err.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json [hunk 2]
```diff
@@ -349,6 +349,152 @@
...
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "204": {
```
Comment (heaths) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json:474]: This should 404, which is also consistent with Key Vault: https://github.com/Azure/azure-rest-api-specs/blob/7256c18a35090d545a5a3595920c2275d3e397b3/specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keyvault.json#L653

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json [hunk 3]
```diff
@@ -349,6 +349,152 @@
...
+            "description": "Request successful. Resource does not exist."
+          },
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously."
+          },
+          "200": {
```
Comment (heaths) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json:480]: As discussed for the data plane, operations shouldn't both return 200 and 202. It creates an unnecessary fork in code. That said, I see that Key Vault also does this already as well. If you're sharing APIs - at least how you are in the data plane - behavior should be consistent. Even here in the control plane, consistency will lead to fewer incorrect assumptions in the CLI and other clients.

/cc @johanste

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json [hunk 4]
```diff
@@ -764,6 +910,12 @@
             "name": "PublicNetworkAccess",
             "modelAsString": true
           }
+        },
+        "purgeDateTimeUtc": {
```
Comment (heaths) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json:914]: 1. All dates must be in UTC anyway, so declaring "Utc" in the property name is unnecessary and not consistent with most other providers I've seen.
2. For consistency's sake with Key Vault, could this just be `scheduledPurgeDate`: https://github.com/Azure/azure-rest-api-specs/blob/7256c18a35090d545a5a3595920c2275d3e397b3/specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keyvault.json#L1410

**Review by heaths — CHANGES_REQUESTED** [review]

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json (hunk 1 above)
Comment (kim-soohwan) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json]: This is complete.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/examples/DeletedManagedHsm_Purge.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
+  "parameters": {
+    "name": "hsm1",
+    "location": "westus",
+    "subscriptionId": "00000000-0000-0000-0000-000000000000",
+    "api-version": "2021-04-01-preview"
+  },
+  "responses": {
+    "200": {},
+    "202": {},
```
Comment (heaths) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/examples/DeletedManagedHsm_Purge.json:10]: 1. As mentioned elsewhere, this shouldn't return both 200 and 202.
2. That said, what would a 202 even mean here anyway? After a deleted vault / HSM is purged, there's no resources. Key Vault's purge - at least in the data plane (so please check the mgmt plane) only returns 200. The operation is an LRO, but there's nothing to poll.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json (hunk 3 above)
Comment (heaths) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json:488]: This still defines 200. It should only return 202 if it's an LRO.

**Review by heaths — CHANGES_REQUESTED** [review]

**Review by heaths — APPROVED** [review]

**Pull request merged**